ResetPasswordBundle/src/Persistence/Repository/PasswordResetRequestRepositoryTrait.php

```diff
@@ -1,58 +1,67 @@
 
-use SymfonyCasts\Bundle\ResetPassword\Model\PasswordResetRequest;
+use Symfony\Component\Security\Core\User\UserInterface;
+use SymfonyCasts\Bundle\ResetPassword\Model\PasswordResetRequestInterface;
 
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ */
 trait PasswordResetRequestRepositoryTrait
 {
-    public function createPasswordResetRequest(object $user, \DateTimeInterface $expiresAt, string $selector, string $hashedToken): PasswordResetRequest
-    {
-        return new PasswordResetRequest(
-            $user,
-            $expiresAt,
-            $selector,
-            $hashedToken
-        );
-    }
-
-    public function getUserIdentifier(object $user): string
+    //@todo implement method or remove it
+    public function getUserIdentifier(UserInterface $user): string
     {
         $this->getEntityManager()
             ->getUnitOfWork()
             ->getSingleIdentifierValue($user);
     }
 
-    public function persistPasswordResetRequest(PasswordResetRequest $passwordResetRequest)
+    public function persistPasswordResetRequest(PasswordResetRequestInterface $passwordResetRequest)
     {
         $this->getEntityManager()->persist($passwordResetRequest);
         $this->getEntityManager()->flush($passwordResetRequest);
     }
 
-    public function findPasswordResetRequest(string $selector): ?PasswordResetRequest
+    public function findPasswordResetRequest(string $selector): ?PasswordResetRequestInterface
     {
         return $this->findOneBy(['selector' => $selector]);
     }
 
-    public function getMostRecentNonExpiredRequestDate(object $user): ?\DateTimeInterface
+    public function getMostRecentNonExpiredRequestDate(UserInterface $user): ?\DateTimeImmutable
     {
         // Normally there is only 1 max request per use, but written to be flexible
-        /** @var PasswordResetRequest[] $resetRequests */
-        $resetRequests = $this->createQueryBuilder('t')
-            ->select('t.requestedAt')
+        /** @var PasswordResetRequestInterface $resetRequest */
+        $resetRequest = $this->createQueryBuilder('t')
             ->where('t.user = :user')
             ->setParameter('user', $user)
             ->orderBy('t.requestedAt', 'DESC')
             ->setMaxResults(1)
             ->getQuery()
-            ->execute();
+            ->getOneorNullResult()
+        ;
 
-        foreach ($resetRequests as $resetRequest) {
-            if (!$resetRequest->isExpired()) {
-                return $resetRequest->getRequestedAt();
-            }
+        if (null !== $resetRequest && !$resetRequest->isExpired()) {
+            return $resetRequest->getRequestedAt();
         }
 
         return null;
     }
+
+    public function removeResetRequest(PasswordResetRequestInterface $resetRequest): void
+    {
+        $this->getEntityManager()->remove($resetRequest);
+        $this->getEntityManager()->flush();
```

Comment:
Does it work to do `$this->getEntityManager()->flush($resetRequest);` when removing objects? I know that works when saving them - but not sure about removal. We want to avoid (if possible) flushing any changes that the user might have "pending" in their own entities.

Reply:
I wasn't aware of this, but calling flush() with any args is deprecated and the ability to do so will be removed in ORM 3.0. Was trying to find more information on this, but looking at the source, 

```
public function flush($entity = null)
    {
        if ($entity !== null) {
            @trigger_error(
                'Calling ' . __METHOD__ . '() with any arguments to flush specific entities is deprecated and will not be supported in Doctrine ORM 3.0.',
                E_USER_DEPRECATED
            );
        }

        $this->errorIfClosed();

        $this->unitOfWork->commit($entity);
    }
```

I think it may be best to leave this as is now, and the same goes for `persistResetPasswordRequest()`, and I'll address both methods in a separate PR. Perhaps come up with a strategy to check the state of the `UnitOfWork` or something.. Else let it ride (after removing the direct `flush(s)`) and leave it up to the consumer to ensure there are not any pending changes before using our reset password..

Reply:
done - not flushing entities directly.. will handle the possible inadvertent flushing of "sloppy" Unit of Work in the future.